### PR TITLE
Adding description for Optimizers

### PR DESCRIPTION
**Pull request by vishwakftw**
This closes issue https://github.com/pytorch/pytorch/issues/4362 . 

Please let me know your views and suggest improvements if any.

### CODE REVIEW

**Comment by soumith** [comment]
lgtm, add a test in test_optim.py that calls repr (dont need to check the return value other than that this is a string). This way we dont break this while doing other optim changes.

File: test/test_optim.py [hunk 1]
```diff
@@ -113,6 +123,9 @@ def _test_basic_cases_template(self, weight, bias, input, constructor):
         input = Variable(input)
         optimizer = constructor(weight, bias)
 
+        # to check if the optimizer can be printed as a string
+        optimizer.__repr__()
```
Comment (apaszke) [inline test/test_optim.py]: It's enough to call this here and test it once. Please remove all other places

**Review by apaszke — CHANGES_REQUESTED** [review]

File: test/test_optim.py (hunk 1 above)
Comment (vishwakftw) [reply test/test_optim.py]: Thank you. This has been done as suggested.

**Review by apaszke — APPROVED** [review]

**Comment by apaszke** [comment]
Thank you!

**Pull request merged**